pkg/apis/resource/types.go

```diff
@@ -0,0 +1,466 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Claim is created by users to describe which resources they need.
+// Its status tracks whether the resource has been allocated and what the
+// resulting attributes are.
+//
+// This is an alpha type and requires enabling the DynamicResourceAllocation
+// feature gate.
+type Claim struct {
```

Comment:
We don't have Go's "don't stutter" rule.  Shouldn't this be `ResourceClaim` ?

Reply:
@alculquicondor suggested it and it made sense to me. I can go back to ResourceClaim, ResourceClass, and ResourceClaimTemplate if you prefer that.

Reply:
Looks like you really do :grin: 

I'll revert back to the longer names.

Reply:
I'm happy to revisit this as a conventions, but we seem to have a general common style of not abbreviating names like this.  Otherwise the YAML `kind` and Go type name don't match and docs get confused.